crates/zk-por-core/src/circuit_registry/registry.rs

```diff
@@ -0,0 +1,167 @@
+
+        let start = std::time::Instant::now();
+        let prover = MerkleSumTreeProver { accounts: accounts };
+        let empty_batch_proof =
+            prover.get_proof_with_circuit_data(account_targets.clone(), &batch_circuit_data);
+        tracing::info!(
+            "prove merkle sum tree with batch size {} in : {:?}",
+            batch_size,
+            start.elapsed()
+        );
+
+        let mut recursive_circuit_and_empty_proofs = Vec::new();
+
+        let mut last_circuit_data = &batch_circuit_data;
+        let mut last_empty_proof = empty_batch_proof.clone();
+
+        for (level, circuit_config) in recursive_level_configs.into_iter().enumerate() {
+            let start = std::time::Instant::now();
+            let (recursive_circuit, recursive_targets) =
+                build_recursive_n_circuit::<C, RECURSIVE_FACTOR>(
+                    &last_circuit_data.common,
+                    &last_circuit_data.verifier_only,
+                    circuit_config,
+                );
+            tracing::info!(
+                "build recursive circuit at level {} in : {:?}, with vd {:?}",
+                level,
+                start.elapsed(),
+                recursive_circuit.verifier_only.circuit_digest
+            );
+            let sub_proofs: [ProofWithPublicInputs<F, C, D>; RECURSIVE_FACTOR] =
+                std::array::from_fn(|_| last_empty_proof.clone());
+            let start = std::time::Instant::now();
+            let recursive_prover = RecursiveProver {
+                sub_proofs: sub_proofs,
+                sub_circuit_vd: last_circuit_data.verifier_only.clone(),
+            };
+            let recursive_proof = recursive_prover
+                .get_proof_with_circuit_data(recursive_targets.clone(), &recursive_circuit);
+
+            tracing::info!(
+                "prove recursive subproofs at level {} in : {:?}",
+                level,
+                start.elapsed()
+            );
+
+            recursive_circuit_and_empty_proofs
+                .push(((recursive_circuit, recursive_targets), recursive_proof.clone()));
+
+            last_circuit_data = &recursive_circuit_and_empty_proofs.last().unwrap().0 .0;
+            last_empty_proof = recursive_proof;
+        }
+
+        tracing::info!(
+            "finish init circuit registry with {} recursive levels in {:?}",
+            recursive_circuit_and_empty_proofs.len(),
+            init_start.elapsed()
+        );
+
+        Self {
+            batch_circuit: (batch_circuit_data, account_targets),
+            empty_batch_proof: empty_batch_proof,
+            recursive_circuits_and_empty_proofs: recursive_circuit_and_empty_proofs,
+        }
+    }
+
+    pub fn get_batch_circuit(&self) -> (&CircuitData<F, C, D>, Vec<AccountTargets>) {
+        (&self.batch_circuit.0, self.batch_circuit.1.clone())
+    }
+
+    pub fn get_empty_batch_circuit_proof(&self) -> ProofWithPublicInputs<F, C, D> {
```

Comment:
this impl should only exist in test env; not release

Reply:
this impl will use release mode, as we need to pad empty proofs.